src/main/java/org/folio/service/organization/OrganizationStorageService.java

```diff
@@ -44,28 +45,29 @@ public class OrganizationStorageService extends BaseService implements Organizat
   private AcquisitionsUnitsService acquisitionsUnitsService;
 
   @Override
-  public CompletableFuture<Organization> createOrganization(Organization organization, Context context,
-      Map<String, String> headers) {
+  public Future<Organization> createOrganization(Organization organization, Context context, Map<String, String> headers) {
     logger.debug("createOrganization:: Trying to create organization with name: {}", organization.getName());
-    HttpClientInterface client = getHttpClient(headers);
-    CompletableFuture<Organization> future = new CompletableFuture<>();
+    RequestContext requestContext = new RequestContext(context, headers);
+    Promise<Organization> promise = Promise.promise();
+
     if (isSameAccountNumbers(organization)) {
       logger.warn("crateOrganization:: Account number of organization '{}' is not unique", organization.getName());
-      future.completeExceptionally(new HttpException(HttpStatus.HTTP_UNPROCESSABLE_ENTITY.toInt(),
+      return Future.failedFuture(new HttpException(HttpStatus.HTTP_UNPROCESSABLE_ENTITY.toInt(),
         ACCOUNT_NUMBER_MUST_BE_UNIQUE.toError()));
-      return future;
     }
-    return handlePostRequest(JsonObject.mapFrom(organization), resourcesPath(ORGANIZATIONS), client, context, headers, logger)
-      .thenApply(id -> JsonObject.mapFrom(organization.withId(id))
-        .mapTo(Organization.class))
-      .handle((org, t) -> {
-        client.closeClient();
-        if (Objects.nonNull(t)) {
-          logger.warn("Error creating organization with name: {}", organization.getName(), t);
-          throw new CompletionException(t.getCause());
-        }
-        return org;
-      });
+    handlePostRequest(JsonObject.mapFrom(organization), resourcesPath(ORGANIZATIONS), requestContext, logger)
```

Comment:
There is no need to convert records into json object and then map to java class again. Just use a proper method in RestClient as it was done in mod-orders
https://github.com/folio-org/mod-orders/blob/master/src/main/java/org/folio/rest/core/RestClient.java#L44

Reply:
fixed